src/sentry/api/serializers/models/team.py

```diff
@@ -184,3 +185,22 @@ def serialize(
         d["projects"] = attrs["projects"]
         d["externalTeams"] = attrs["externalTeams"]
         return d
+
+
+class TeamSCIMSerializer(Serializer):  # type: ignore
+    def serialize(
+        self, obj: Team, attrs: Mapping[str, Any], user: Any, **kwargs: Any
+    ) -> MutableMapping[str, JSONData]:
+        return {
+            "schemas": [SCIM_SCHEMA_GROUP],
+            "id": obj.id,
+            "displayName": obj.slug,
+            # For a patch request, we don't need to return the full list of members
+            # so look at exclude_members arg
+            "members": [
+                {"value": str(om.id), "display": om.get_email()} for om in obj.member_set
+            ]  # TODO: what happens when this gets very large?
```

Comment:
it does not. it seems that this "members" attribute is always excluded for azure, and in some cases for okta it can be omitted. don't see any mentions of this being a problem googling, but i imagine having it be unbounded isn't ideal. will ask okta support

Reply:
I think the main thing here is that you might want to paginate over `member_set` so that you don't have a large number of models in memory at once. 20k id/email dicts is probably fine, but 20k models takes up a lot more memory. You can use `RangeQuerySetWrapper` to handle this. 

I also noticed that `OrganizationMember.get_email` will access `.user.email`, so you want to use `select_related("user")` here to avoid N queries on the user table.

Reply:
👍🏼 @wedamija waiting for okta support to get back to me on if this field a hard requirement, if so will implement your suggestions, if not, will just omit the members list.

Reply:
@JoshFerge Why'd we merge this without addressing this point?

Reply:
@wedamija sorry, didn't mean to leave this hanging. i got back a really wishy washy answer from okta support and need to do further research on what other implementations do, and didn't want to leave the PR stale for this. I'll be making a few follow up PRs in the next few days / weeks addressing this and some other optimizations. Let me know if you have questions/concerns.

Reply:
Sounds good, thanks!